### DISCUSSION TITLE
Search

### DISCUSSION THREAD

**Discussion by bwaltz** [ColbyCommunications/colby-college-theme#45] [General]
@gregoryspragginsjr I'm trying something out with the search on my local. Does the search.vue component know when the modal is open vs when it's closed? I notice that you are sending a `showModal` prop to the slot which will contain the search. Does that mean `showModal` is accessible in search.vue? If so, how?

**Comment by gregoryspragginsjr** [comment]
@bwaltz Sort of! Let me explain. I am using an event emitter here via one of my favorite little packages for Vue, [Mitt](https://www.npmjs.com/package/mitt/v/1.0.1). Mitt is currently firing an event called `close-modal` within `Modal.vue` that `Search.vue` is listening out for and triggering a function in response. That is how we get the search query in the Search modal to reset every time time you close the modal.

If you're looking to build on this, and do something interesting everytime the Modal is closed, all you need to do is write something like:

```
this.emitter.on('close-modal', () => {
  // DO SOMETHING
});
```

To further clarify, `showModal` in this context of the slots is not being used for anything (I should probably remove it from those slots on lines 42 and 31). If you'd like an example of passing data through slots, I'd take a look at `Carousel.vue` and `carousel.twig`

**Comment by bwaltz** [comment]
So in search.vue how would I do `v-if="modalOpen"`? Is there something passed to that (or can be passed) that I can utilize?

**Comment by gregoryspragginsjr** [comment]
@bwaltz You'd have two options:

- You could try to utilize `showModal` being passed through the slots
- I would amend `Modal.vue` line 76 to fire a new `opened-modal` event. You would then wire up some action in `Search.vue` `mounted()` (specifically line 110) to then do something whenever `opened-modal` is fired.
